<launch>
    <arg name="output"              default="screen"                doc="display output to screen or log file"/>
    <arg name="odom_topic"          default="/slam_pose"      doc="ROS topic for the pose"/>
    <arg name="control_topic"       default="/Control"              doc="ROS topic for control input"/>
    <arg name="obstacle_topic"      default="/Prediction/Obstacles" doc="ROS topic for obstacles"/>
    <arg name="pub_rate"            default="20"                    doc="rate at which to publish the pose"/>
    <arg name="init_x"              default="2.5"                     doc="initial x position"/>
    <arg name="init_y"              default="0.15"                  doc="initial y position"/>
    <arg name="init_yaw"            default="0"                     doc="initial z position"/>
    <arg name="lane_change_cost"    default="2"                     doc="cost of lane change"/>



    <arg name="num_dyn_obs"         default="2"                     doc="number of dynamic obsects"/>
    <arg name="num_static_obs"      default="0"                     doc="number of static obsects"/>
    <arg name="static_obs_size"     default="0.18"                   doc="type of static obsects"/>
    <arg name="static_obs_topic"    default="/Obstacles/Static"     doc="ROS topic for static obstacles"/>
    <arg name="dyn_obs_topic"       default="/Obstacles/Dynamic"    doc="ROS topic for dynamic obstacles"/>


    <!-- TODO: add your decision making and trajectory planning nodes here -->
    <!-- This section launch the simulation of the student truck -->
    
    <include file="$(find racecar_interface)/launch/simulation.launch" >
        <arg name="odom_topic"              value="$(arg odom_topic)" />
        <arg name="control_topic"           value="$(arg control_topic)" />
        <arg name="pub_rate"                value="$(arg pub_rate)" />
        <arg name="init_x"                  value="$(arg init_x)" />
        <arg name="init_y"                  value="$(arg init_y)" />
        <arg name="init_yaw"                value="$(arg init_yaw)" />
        <arg name="lane_change_cost"        value="$(arg lane_change_cost)"/>
    </include>

<!-- <include file="$(find racecar_interface)/launch/traffic_simulation.launch" >
        <arg name="odom_topic"              value="$(arg odom_topic)" />
        <arg name="control_topic"           value="$(arg control_topic)" />
        <arg name="pub_rate"                value="$(arg pub_rate)" />
        <arg name="init_x"                  value="$(arg init_x)" />
        <arg name="init_y"                  value="$(arg init_y)" />
        <arg name="init_yaw"                value="$(arg init_yaw)" />
        <arg name="enable_routing"          value="true" />
        <arg name="lane_change_cost"        value="$(arg lane_change_cost)"/>
        <arg name="num_dyn_obs"             value="$(arg num_dyn_obs)"/>
        <arg name="num_static_obs"          value="$(arg num_static_obs)"/>
        <arg name="static_obs_size"         value="$(arg static_obs_size)"/>
        <arg name="static_obs_topic"        value="$(arg static_obs_topic)"/>
        <arg name="dyn_obs_topic"           value="$(arg dyn_obs_topic)"/>
    </include> -->





    <include file="$(find racecar_obs_detection)/launch/frs.launch" >
       
    </include>
    

    <node pkg="final_project" type="task2_controller_nodewrapper.py" name="task2_controller" output="$(arg output)">
    
    </node>
   
    <node pkg="racecar_planner" type="traj_planning_node.py" name="traj_planning" output="$(arg output)">
        <param name="odom_topic"           value="$(arg odom_topic)" />
        <param name="control_topic"        value="$(arg control_topic)" />
        <param name="obstacle_topic"       value="$(arg obstacle_topic)" />
        <param name="package_path"         value="$(find racecar_planner)" />
        <param name="simulation"           value="False" />
        <param name="replan_dt"            value="0.1" />
        <param name="receding_horizon"     value="true" />
        <param name="ilqr_params_file"     value="/mnt/c/Users/miche/OneDrive/repos/ECE346_OsaAnna/ROS_Core/src/Labs/Lab1/configs/lab2_task2.yaml" />
        <param name="PWM_model"            value="$(find racecar_planner)/configs/mlp_model.sav" />
    </node>

    <!-- This section launch the simulation of the boss truck -->
    <!-- You DO NOT need to modify this section -->
    <node pkg="racecar_interface" type="simulation_node.py" name="boss_simulation_node" output="$(arg output)">
        <param name="odom_topic"           value="/Boss/Pose" />
        <param name="control_topic"        value="/Boss/servo_control" />
        <param name="pub_rate"             value="$(arg pub_rate)" />
        <param name="init_x"               value="3.5" />
        <param name="init_y"               value="0.15" />
        <param name="init_yaw"             value="0" />
        <param name="serice_name"          value="Boss/reset" />
    </node>

    <!-- This section launch the world model -->
    <!-- You DO NOT need to modify this section -->
    <node pkg="final_project" type="task2_world_node.py" name="task2_world_node" output="$(arg output)">
        <param name="time_limit"                value="300" />
        <param name="dt_refresh"                value="5" />
        <param name="warehouse_yaml"            value="$(find final_project)/task2.yaml" />
        <param name="student_pose_topic"        value="$(arg odom_topic)" />
        <param name="PWM_model"                 value="$(find final_project)/pwm.sav" />
        <param name="boss_odom_topic"           value="/Boss/Pose" />
        <param name="boss_control_topic"        value="/Boss/servo_control" />
        <param name="simulation"                value="False" />
    </node>

</launch>
